### PR TITLE
Add use case for searching persons

### PR DESCRIPTION
**Pull request by Javiery3889**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -492,6 +492,36 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+  * 1a1. FINDvisor displays an empty list.
+
+    Use case ends.
+
+* 1b. FINDvisor detects an error in specified keywords.
+  * 1b1. FINDvisor shows an error message and requests for valid keywords from the user.
+  * 1b2. User enters new keywords along with search category.
+  * 1c3. Steps 1b1 - 1b2 are repeated until the keywords given by the user is valid.
+
+    Use case ends.
```
Comment (Dethada) [inline docs/DeveloperGuide.md:516]: If we follow our current use case format, I think this should be
```suggestion
* 1b. FINDvisor detects an error in specified keywords.
  * 1b1. FINDvisor shows an error message and requests for valid keywords from the user.

    Use case resumes at step 1.
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -492,6 +492,36 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+
+* 1c. The given category is invalid.
+  * 1c1. FINDvisor shows an error message and requests for a valid category from the user.
+  * 1c2. User enters new category with keywords.
+  * 1c3. Steps 1c1 - 1c2 are repeated until the category given by the user is valid.
+
+    Use case resumes at step 2.
```
Comment (Dethada) [inline docs/DeveloperGuide.md:523]: Same comment as above

**Review by Dethada — COMMENTED** [review]
Comments about following our current use case format

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -492,6 +492,36 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
 
     Use case resumes at step 2.
 
+**Use case: Search for specific persons based on a person's field**
```
Comment (KohGuanZeh) [inline docs/DeveloperGuide.md:495]:
```suggestion
#### Use case: Search for specific persons based on a person's field
```
I think we should use heading-4 for this for better readability?

**Review by KohGuanZeh — DISMISSED** [review]
LGTM

**Review by Arixeyeion — APPROVED** [review]

**Review by KohGuanZeh — APPROVED** [review]

**Pull request merged**